engine/crates/parser-sdl/src/rules/federation/key.rs

```diff
@@ -0,0 +1,116 @@
+use engine::registry::federation::Selection;
+use serde::Deserializer;
+
+use crate::rules::directive::Directive;
+
+#[derive(Debug, serde::Deserialize)]
+#[serde(rename_all = "camelCase", deny_unknown_fields)]
+pub struct KeyDirective {
+    #[serde(deserialize_with = "deserialize_selections")]
+    pub fields: Vec<Selection>,
+    #[serde(default = "default_to_true")]
+    pub resolvable: bool,
+}
+
+fn default_to_true() -> bool {
+    true
+}
+
+impl Directive for KeyDirective {
+    fn definition() -> String {
+        // Note: technically this is meant to be declared "repeatable"
+        // but our parser doesn't seem to support it.
+        r#"
+        directive @key(fields: FieldSet!, resolvable: Boolean = true) on OBJECT | INTERFACE
+
+        "#
+        .to_string()
+    }
+}
+
+fn deserialize_selections<'de, D>(deserializer: D) -> Result<Vec<Selection>, D::Error>
+where
+    D: Deserializer<'de>,
+{
+    struct Visitor;
+    impl<'de> serde::de::Visitor<'de> for Visitor {
+        type Value = Vec<Selection>;
+
+        fn visit_str<E>(self, value: &str) -> Result<Self::Value, E>
+        where
+            E: serde::de::Error,
+        {
+            // This whole implementation is extremely naive and doesn't support a lot of stuff.
+            // Will fix in https://linear.app/grafbase/issue/GB-5086
+            if value.contains('{') {
+                return Err(E::custom("nested fields in keys aren't supported at the moment"));
+            }
+
+            Ok(value
+                .split(' ')
```

Comment:
Naive question: I haven't seen this in the unit tests but I guess the expected format is `@key(fields: "a b c")`?

Reply:
To be more accurate: it's a GraphQL query without arguments or directives.  So `@key(fields: "a b { c { d }}")` would be valid.  Though this implementation only supports `a b`.

Reply:
Oooh, I see. Technically commas would be allowed too, then? Probably not relevant to implement at this stage.

Reply:
Yeah, I would assume so.  Going to write/use an actual parser when I revisi the mentioned issue.